tests/functional/object/mcg/ui/test_mcg_ui.py

```diff
@@ -441,3 +442,207 @@ def test_bucket_create(self, setup_ui_class_factory):
+                logger.info(f"Additional page bucket count: {len(next_page_buckets)}")
+                all_ui_buckets.extend(next_page_buckets)
+
+                if len(all_ui_buckets) >= len(cli_buckets):
+                    break
+
+        logger.info(f"Total UI bucket count across all pages: {len(all_ui_buckets)}")
+        logger.info(f"CLI bucket count: {len(cli_buckets)}")
+
+        ui_bucket_set = set(all_ui_buckets)
+        cli_bucket_set = set(cli_buckets)
+
+        missing_in_ui = cli_bucket_set - ui_bucket_set
+        if missing_in_ui:
+            logger.warning(f"Buckets in CLI but missing in UI: {missing_in_ui}")
+
+        missing_in_cli = ui_bucket_set - cli_bucket_set
+        if missing_in_cli:
+            logger.warning(f"Buckets in UI but missing in CLI: {missing_in_cli}")
+
+        assert (
+            len(all_ui_buckets) >= 2
+        ), "Expected at least 2 buckets (OBC and S3) but found less"
+
+        assert len(all_ui_buckets) == len(
+            cli_buckets
+        ), f"UI bucket count ({len(all_ui_buckets)}) does not match CLI bucket count ({len(cli_buckets)})"
+
+    def test_bucket_pagination(self, setup_ui_class_factory, mcg_obj):
+        """
+        Test bucket pagination functionality in UI.
+
+        Steps:
+        1. Navigate to the Object Storage Buckets page
+        2. Check initial bucket count via CLI
+        3. Create additional buckets if needed to exceed 100 total buckets
+        4. Verify pagination controls appear with >100 buckets
+        5. Collect buckets listed on first page (should be 100)
+        6. Navigate to next page using pagination controls
+        7. Collect buckets on second page and verify they differ from first page
+        8. Navigate back to previous page using pagination controls
+        9. Verify the returned page matches the original first page
+        """
+        setup_ui_class_factory()
+        bucket_ui = BucketsTab()
+        bucket_ui.nav_object_storage_page()
+        bucket_ui.nav_buckets_page()
+
+        cli_buckets = list(mcg_obj.cli_list_all_buckets())
+        initial_bucket_count = len(cli_buckets)
+        logger.info(f"Initial bucket count (from CLI): {initial_bucket_count}")
+
+        ui_buckets = bucket_ui.get_buckets_list()
+        ui_bucket_count = len(ui_buckets)
+        logger.info(f"UI bucket count (current page): {ui_bucket_count}")
+
+        # Determine if we need to create more buckets
+        buckets_needed = max(0, 101 - initial_bucket_count)
+        if buckets_needed > 0:
+            logger.info(
+                f"Creating {buckets_needed} additional buckets to test pagination"
+            )
+
+            s3_buckets_to_create = buckets_needed // 2
+            obc_buckets_to_create = buckets_needed - s3_buckets_to_create
+
+            bucket_ui.create_multiple_buckets_ui(
+                s3_buckets=s3_buckets_to_create, obc_buckets=obc_buckets_to_create
+            )
```

Comment:
A suggestion In case this takes too long - use the `bucket_factory` fixture instead to create the buckets via boto3:
```
new_buckets = bucket_factory(100)
```

Reply:
i want to create high amount of buckets via ui

Reply:
NP now that this test is now a tier2 - resolving this thread